src/Nethermind/Nethermind.Blockchain.Test/Synchronization/EthSyncPeerPoolTests.cs

```diff
@@ -552,5 +552,36 @@ public void Does_not_fail_when_receiving_a_new_block_and_allocation_has_no_peer(
 
             _blockTree.NewHeadBlock += Raise.EventWith(new object(), new BlockEventArgs(Build.A.Block.WithTotalDifficulty(1).TestObject));
         }
+        
+        [Test]
+        public async Task Can_borrow_async_many()
+        {
+            var peer = new SimpleSyncPeerMock(TestItem.PublicKeyA);
+            var peer2 = new SimpleSyncPeerMock(TestItem.PublicKeyB);
+
+            _pool.Start();
+            _pool.AddPeer(peer);
+            _pool.AddPeer(peer2);
+            await Task.Delay(200);
+
+            var allocation1 = _pool.BorrowAsync(BorrowOptions.None, string.Empty, null, 1000);
```

Comment:
Can you use collections and collections assertions here? Now its quite not readable

Reply:
where? we never assert on collections content here?

Reply:
ah ok, I see

Reply:
let me fix it

Reply:
done, thanks for pointing out